pkg/cluster/k8sres.go

```diff
@@ -1561,11 +1561,17 @@ func (c *Cluster) generateSingleUserSecret(namespace string, pgUser spec.PgUser)
 	}
 
 	username := pgUser.Name
+	lbls := c.labelsSet(true)
+
+	if username == "pooler" {
```

Comment:
Please use the `ConnectionPoolerUserName` constant from `pkg/util/constants/pooler.go` and not the hard-coded value.